deps/chakrashim/src/jsrtcontextshim.cc

```diff
@@ -482,76 +506,69 @@ void ContextShim::RunMicrotasks() {
   }
 }
 
-JsValueRef ContextShim::GetUndefined() {
-  return undefinedRef;
-}
-
-JsValueRef ContextShim::GetTrue() {
-  return trueRef;
-}
-
-JsValueRef ContextShim::GetFalse() {
-  return falseRef;
-}
-
-JsValueRef ContextShim::GetNull() {
-  return nullRef;
-}
-
-JsValueRef ContextShim::GetZero() {
-  return zero;
-}
-
-JsValueRef ContextShim::GetObjectConstructor() {
-  return globalConstructor[GlobalType::Object];
-}
-
-JsValueRef ContextShim::GetBooleanObjectConstructor() {
-  return globalConstructor[GlobalType::Boolean];
-}
-
-JsValueRef ContextShim::GetNumberObjectConstructor() {
-  return globalConstructor[GlobalType::Number];
-}
-
-JsValueRef ContextShim::GetStringObjectConstructor() {
-  return globalConstructor[GlobalType::String];
-}
-
-JsValueRef ContextShim::GetDateConstructor() {
-  return globalConstructor[GlobalType::Date];
-}
-
-JsValueRef ContextShim::GetProxyConstructor() {
-  return globalConstructor[GlobalType::Proxy];
-}
+// check initialization state first instead of calling
+// InitializeCurrentContextShim to save a function call for cases where
+// contextshim is already initialized
+#define DECLARE_GETOBJECT(name, object) \
+JsValueRef ContextShim::Get##name() { \
+  if(!(initialized || initializing)) { \
+    if (!InitializeCurrentContextShim()) { \
+      Fatal("Failed to initialize current context"); \
+    } \
+  } \
+  return object; \
+} \
 
+DECLARE_GETOBJECT(True, trueRef)
+DECLARE_GETOBJECT(False, falseRef)
+DECLARE_GETOBJECT(Undefined, undefinedRef)
+DECLARE_GETOBJECT(Null, nullRef)
+DECLARE_GETOBJECT(Zero, zero)
+DECLARE_GETOBJECT(ProxyOfGlobal, proxyOfGlobal)
+DECLARE_GETOBJECT(ReflectObject, reflectObject)
+DECLARE_GETOBJECT(ObjectConstructor,
+                         globalConstructor[GlobalType::Object])
```

Comment:
Maybe unindent this line to align with above will look better? Same for the rest.